components/authz-service/engine/opa/opa_internal_test.go

```diff
@@ -313,70 +312,77 @@ func BenchmarkInitPartialResultV2(b *testing.B) {
 	require.NoError(b, err, "init logger")
 	store := inmem.NewFromReader(f)
 
-	for d := 0; d < 5; d++ {
-		b.Run(fmt.Sprintf("default policies, run %d times", d), func(b *testing.B) {
-			s, err := New(ctx, l)
-			require.NoError(b, err, "init state")
-			s.v2Store = store
+	b.Run("default policies", func(b *testing.B) {
+		s, err := New(ctx, l)
+		require.NoError(b, err, "init state")
+		s.v2Store = store
 
-			for n := 0; n < b.N; n++ {
-				for e := 0; e <= d; e++ {
-					r = s.initPartialResultV2(ctx)
-					if r != nil {
-						b.Error(r)
-					}
-				}
+		for n := 0; n < b.N; n++ {
+			r = s.initPartialResultV2(ctx)
+			if r != nil {
+				b.Error(r)
 			}
-			errResult = r
-		})
-	}
+		}
+		errResult = r
+	})
 
-	for k := 0; k < 5; k++ {
-		m := k * 25
-		store = storeWithDummyPolicies(b, m)
-		b.Run(fmt.Sprintf("%d dummy policies", m), func(b *testing.B) {
-			for n := 0; n < b.N; n++ {
-				s, err := New(ctx, l)
-				require.NoError(b, err, "init state")
-				s.v2Store = store
+	s, err := New(ctx, l)
+	require.NoError(b, err, "init state")
+	s.v2Store = store
 
-				r = s.initPartialResultV2(ctx)
-				if r != nil {
-					b.Error(r)
-				}
-			}
-			errResult = r
-		})
+	r = s.initPartialResultV2(ctx)
+	if r != nil {
+		b.Error(r)
 	}
-}
 
-// helpers
+	b.Run("IsAuthorized(true) with the result of BenchmarkInitPartialResultV2", func(b *testing.B) {
```

Comment:
The following three benchmarks allow us to get a grip on how the query performance changes in this PR. (I've cherry-picked the commit onto master to have a comparison.)